### PR TITLE
Cmake integration

### PR DESCRIPTION
**Pull request by fixgoats**
Pull request for issue #120

### CODE REVIEW

**Comment by kratsg** [comment]
We're going to need to set up a GitHub actions to run the cmake configuration so we know that things work well. If @fixgoats is familiar enough with GHA to do this, then feel free to add it in this PR. Otherwise, either @matthewfeickert or myself can do it in a follow-up PR.

File: README.md [hunk 1]
```diff
@@ -50,6 +50,11 @@ cd src
 make
 cd ..
 ```
+or with cmake:
+```
+cmake src -B lib -DCMAKE_BUILD_TYPE=Release
```
Comment (matthewfeickert) [inline README.md:55]: It is great to see some modern CMake here, but especially for docs I would make all args explicit.
```suggestion
cmake -S src -B lib -DCMAKE_BUILD_TYPE=Release
```

File: README.md [hunk 2]
```diff
@@ -50,6 +50,11 @@ cd src
...
+cmake --build lib
```
Comment (matthewfeickert) [inline README.md:56]:
```suggestion
cmake --build lib --clean-first --parallel 8
```

Some optional flags I find quite useful. I normally like to use `--clean-first --parallel $(($(nproc) - 1))` but that might be a bit extreme for some people, so `8` or less is probably fine.

**Comment by fixgoats** [comment]
Matthew's suggestions have been implemented. I'm a beginner at GHA and there's a lot of stuff going on in the workflows file, so I'll leave it to you guys to set the actions up.

**Pull request merged**